### PR TITLE
Make freeze-related calls `inline`

### PR DESCRIPTION
**Pull request by colriot**
This will skip unnecessary method calls for essentially-aliased or constant methods. Also will allow JVM/JS compiler to eliminate unreachable code branches

### CODE REVIEW

File: utils/src/nativeCommonMain/kotlin/com/badoo/reaktive/utils/Freeze.kt [hunk 1]
```diff
@@ -4,12 +4,13 @@ import kotlin.native.concurrent.ensureNeverFrozen
 import kotlin.native.concurrent.freeze
 import kotlin.native.concurrent.isFrozen
 
-actual fun <T> T.freeze(): T = freeze()
+@Suppress("NOTHING_TO_INLINE")
+actual inline fun <T> T.freeze(): T = freeze()
 
 actual fun <T: Any> T.ensureNeverFrozen(): T {
```
Comment (colriot) [inline utils/src/nativeCommonMain/kotlin/com/badoo/reaktive/utils/Freeze.kt]: Probably this one is worth becoming `inline`d too

Comment (arkivanov) [reply utils/src/nativeCommonMain/kotlin/com/badoo/reaktive/utils/Freeze.kt]: Yeah, let's do it. Also inline `isFrozen` please.

Comment (arkivanov) [reply utils/src/nativeCommonMain/kotlin/com/badoo/reaktive/utils/Freeze.kt]: And better to annotate `expect` functions as inline as well

Comment (colriot) [reply utils/src/nativeCommonMain/kotlin/com/badoo/reaktive/utils/Freeze.kt]: `isFrozen` is already inlined. Agree that need to move modifier to `expect` functions, thought about after pushing the PR too

**Comment by colriot** [comment]
Updated with suggested changes

**Pull request merged**